packages/core/src/Graphs/Graph.ts

```diff
@@ -30,21 +31,15 @@ export class Graph {
         `can not create new node of type ${nodeTypeName} with id ${nodeId} as one with that id already exists.`
       );
     }
-    let nodeDescription = undefined;
-    if (this.registry.nodes.contains(nodeTypeName)) {
-      nodeDescription = this.registry.nodes.get(nodeTypeName);
-    }
-    if (nodeDescription === undefined) {
-      throw new Error(
-        `no registered node descriptions with the typeName ${nodeTypeName}`
-      );
-    }
-    const node = nodeDescription.factory(
-      nodeDescription,
-      this,
-      nodeConfiguration
+
+    const node = createNodeUsingRegistryDefinition(
```

Comment:
this function is now defined statically, making it more easily testable and callable from the outside.